src/types/ErrorRate/__tests__/utils.test.ts

```diff
@@ -0,0 +1,77 @@
+import * as U from '../';
+import { ToleranceConfig } from '../../ServerConfig';
+
+const defaultTolerance = (): ToleranceConfig => {
+  return {
+    code: new RegExp(''),
+    degraded: U.DEFAULTCONF.degraded,
+    failure: U.DEFAULTCONF.failure,
+    protocol: U.DEFAULTCONF.protocol,
+    direction: U.DEFAULTCONF.direction
+  };
+};
+
+describe('Utils library', () => {
+  describe('Parse Annotation', () => {
+    it('should return tolerances with defaults if only code', () => {
+      var tol = defaultTolerance();
+      tol.code = new RegExp('[45]');
+      expect(U.parseAnnotationTEST('[45]')).toEqual(tol);
+    });
+
+    it('should replace x or X by d', () => {
+      var tol = defaultTolerance();
+      tol.code = new RegExp('[45]d[2]');
```

Comment:
This RegExp:
```
tol.code = new RegExp('[45]d[2]');
```
won't match any "442" I guess it should be 

```
tol.code = new RegExp('[45]\\d[2]');
```

Instead, and from here propagate this in places where used, I'm not finished with the review but I was stopped by this one.

(Unless I'm missing any other transformation out of the scope of that test).